docs/it-manual/api/program-applications.md

```diff
@@ -1,182 +1,781 @@
+
+#### `ti_organization`
+- **Property**: `ti_organization`
+- **JSON Type**: `string` or `null`
+- **Value**: All characters.
+- **Description**: The name of the organization the TI belongs to, as specified in the Intermediaries admin panel. See [Manage trusted intermediaries](/docs/user-manual/civiform-admin-guide/manage-trusted-intermediaries.md) for more.
+
+#### `application`
+- **Property**: `application`
+- **JSON Type**: `object`
+- **Value**: The applicant's application
+- **Description**: See the [Application object](#application-object) section below.
+
+## Application object
+The `application` object contains the applicant's answers to the questions in the program. It's a snapshot of the applicant's answers at the time they clicked submit on their application, and is not updated if they change their answer to a question as part of filling out an application to a different program.
+
+If they update their answers to _this_ program and click submit again, a new application entry is created. See the [`revision_state`](#revision_state) metadata field for more.
+
+The `application` object includes all questions that are in a program, even if the applicant did not answer them or they were skipped due to a visibility condition.
+
+{% hint style="warning" %}
+**Warning** Currently only the questions that are in the most recent version of the program are included in the `application` object. Follow [Github Issue #5018](https://github.com/civiform/civiform/issues/5018) for the work to include all questions that have ever been part of a program in the `application` object.
+{% endhint %}
+
+The `application` object is organized as a set of `key`-`object` pairs that represent each question in the program. The `key` is generated from the admin name of the question, and the `object`contains the applicant's answer. For example, this is the `application` object for a program with two questions:
+- "What is your name?", which has a question key of `applicant_name`.
+- "What is your favorite color?", which has a question key of `favorite_color`.
+
+{% code %}
+```json
 {
-  program_name: string,
-  program_version_id: integer,
-  application_id: integer,
-  language: string,
-  submitter_email: string,
-  application: object
+  "applicant_name": {
+    "question_type": "NAME",
+    "first_name": "Taylor",
+    "middle_name": "Allison",
+    "last_name": "Swift",
+  },
+  "favorite_color": {
+    "question_type": "TEXT",
+    "text": "purple"
+  }
 }
 ```
+{% endcode %}
+
+### Question keys
+The question key is generated from the immutable admin name of the question. The immutable admin name of the question is specified by the admin when creating a question, and the question key is visible in the [generated API docs](#generated-program-specific-docs) page.
+
+The question key is generated by removing non-letter characters and replacing spaces with underscores, so `applicant name4?` becomes `applicant_name`.
```

Comment:
question (non-blocking): Is it possible for admin to name their questions something like "applicant name4" and "applicant name5"? Would that result in two identical question keys in the API?

Reply:
That's caught by validation when the question is created. The validation would disallow the second `applicant name5` name because it maps to the same key as the already existing `applicant name4`.